prow/resultstore/payload.go

```diff
@@ -151,37 +160,68 @@ func (p *Payload) invocationProperties() []*resultstore.Property {
 			Value: p.Job.Status.URL,
 		},
 	}
-	if p.Job.Spec.Refs != nil {
-		ps = append(ps, &resultstore.Property{
-			Key:   "Repo",
-			Value: p.Job.Spec.Refs.RepoLink,
-		})
-	}
-	if p.Started != nil {
-		ps = append(ps, &resultstore.Property{
-			Key:   "Commit",
-			Value: p.Started.RepoCommit,
-		})
-		for _, branch := range p.Started.Repos {
-			ps = append(ps, &resultstore.Property{
-				Key:   "Branch",
-				Value: branch,
-			})
-		}
+	ps = append(ps, p.podSpecProperties()...)
+	ps = append(ps, p.startedProperties()...)
+	return ps
+}
+
+func (p *Payload) podSpecProperties() []*resultstore.Property {
+	var ps []*resultstore.Property
+	if p.Job.Spec.PodSpec == nil {
+		return ps
 	}
-	if p.Job.Spec.PodSpec != nil {
-		for _, c := range p.Job.Spec.PodSpec.Containers {
-			for _, e := range c.Env {
+	seenEnv := map[string]bool{}
+	for _, c := range p.Job.Spec.PodSpec.Containers {
+		for _, e := range c.Env {
+			v := e.Name + "=" + e.Value
+			if !seenEnv[v] {
```

Comment:
Won't this check break if the inputs have the empty strings for `e.Name` and `e.Value`? If empty strings are guaranteed to never happen, then it's worth commenting in here to explain what is responsible for that.

Reply:
Done.